skrf/network.py

```diff
@@ -6073,7 +6074,7 @@ def connect_s(A: np.ndarray, k: int, B: np.ndarray, l: int, num: int = 1) -> np.
     nC = nA + nB  # num ports on C
 
     # create composite matrix, appending each sub-matrix diagonally
-    C = np.zeros((nf, nC, nC), dtype='complex')
+    C = np.zeros((nf, nC, nC), dtype='complex', order='F')
```

Comment:
could we get an issue if return `C` with this modification? I got an error about incompatible memory layout in numpy once and I'm not sure if this change could trigger this behaviour.

Reply:
> could we get an issue if return `C` with this modification? I got an error about incompatible memory layout in numpy once and I'm not sure if this change could trigger this behaviour.

Specifying `order='F'` for the memory layout seems safe since `C` is just an `ndarray` for **storing the assembled S-parameters**. When passed to `innerconnect_s()` (renamed `A`), only the `slicing`/`advanced indexing` result of `A` are used for computations, not `A` itself.

In the `connect_s()` method, the assembled matrix `C` doesn't directly participate in computations. So, optimizing assignment performance with `order='F'` should be harmless. What do you think?

Reply:
Then it should be perfectly fine.